base/compiler/ssair/slot2ssa.jl

```diff
@@ -153,10 +153,17 @@ function fixemup!(cond, rename, ir::IRCode, ci::CodeInfo, idx::Int, @nospecializ
                 return nothing
             end
             op[] = x
-        elseif isa(val, GlobalRef) && !(isdefined(val.mod, val.name) && isconst(val.mod, val.name)) ||
-               (isa(val, Expr) && val.head === :static_parameter)
+        elseif isa(val, GlobalRef) && !(isdefined(val.mod, val.name) && isconst(val.mod, val.name))
             op[] = NewSSAValue(insert_node!(ir, idx,
                 NewInstruction(val, typ_for_val(val, ci, ir.sptypes, idx, Any[]))).id - length(ir.stmts))
+        elseif isexpr(val, :static_parameter)
+            ty = typ_for_val(val, ci, ir.sptypes, idx, Any[])
+            if isa(ty, Const)
+                inst = NewInstruction(quoted(ty.val), ty)
+            else
+                inst = NewInstruction(val, ty)
+            end
+            op[] = NewSSAValue(insert_node!(ir, idx, inst).id - length(ir.stmts))
```

Comment:
Or we may want to do this inlining in the main inlining algorithm (maybe within `process_simple!`)?